identity-wallet/src/state/trust_list/reducers/trust_list_edit.rs

```diff
@@ -0,0 +1,73 @@
+use crate::error::AppError;
+use crate::state::trust_list::actions::trust_list_edit::TrustListsEdit;
+use crate::state::trust_list::TrustList;
+use crate::state::{
+    actions::{listen, Action},
+    AppState,
+};
+
+// todo: remove expect
+pub async fn trust_list_edit(state: AppState, action: Action) -> Result<AppState, AppError> {
+    if let Some(action) = listen::<TrustListsEdit>(action) {
+        let mut trust_lists = state.trust_lists.clone();
+        let trust_list = trust_lists
+            .get_mut(&action.trust_list_id)
+            .expect("error: incorrect trust_list_id dispatched by frontend")
+            .clone();
+
+        trust_lists.remove(&action.trust_list_id);
+
+        trust_lists.insert(TrustList {
+            id: trust_list.id,
+            display_name: action.new_display_name,
+            custom: trust_list.custom,
+            entries: trust_list.entries,
+        });
```

Comment:
No need to:
1. first: clone the `trust_list`
2. then: use its ID to remove it from the `trust_lists`
3. and then: insert a new trust list.

You're already getting a mutable reference to the `trust_list` so you can then immediately update its display name (including error handling instead of the `expect`):

```rust
        let trust_list = trust_lists
            .get_mut(&action.trust_list_id)
            .ok_or_else(|| AppError::Error("incorrect trust_list_id dispatched by frontend".to_string()))?;

        trust_list.display_name = action.new_display_name;
```

Reply:
fixed